plugins/modules/foreman_hostgroup.py

```diff
@@ -340,20 +337,7 @@ def main():
         if 'locations' in entity_dict:
             entity_dict['locations'] = module.find_resources_by_title('locations', entity_dict['locations'], thin=True)
 
-        if 'compute_resource' in entity_dict:
-            entity_dict['compute_resource'] = module.find_resource_by_name('compute_resources', name=entity_dict['compute_resource'], failsafe=False, thin=True)
-
-        if 'compute_profile' in entity_dict:
-            entity_dict['compute_profile'] = module.find_resource_by_name('compute_profiles', name=entity_dict['compute_profile'], failsafe=False, thin=True)
-
-        if 'domain' in entity_dict:
-            entity_dict['domain'] = module.find_resource_by_name('domains', name=entity_dict['domain'], failsafe=False, thin=True)
-
-        if 'subnet' in entity_dict:
-            entity_dict['subnet'] = module.find_resource_by_name('subnets', name=entity_dict['subnet'], failsafe=False, thin=True)
-
-        if 'subnet6' in entity_dict:
-            entity_dict['subnet6'] = module.find_resource_by_name('subnets', name=entity_dict['subnet6'], failsafe=False, thin=True)
+        module.handle_common_host_spec(entity_dict)
```

Comment:
I wrote that on IRC, but I'll repeat here so that others can comment too. This modifies `entity_dict` in-place and is kinda not obvious to the user. I wonder if `entity_dict = module.handle_common_host_spec(entity_dict)` would be more explicit?